src/OrleansRuntime/GrainDirectory/GrainDirectoryPartition.cs

```diff
@@ -124,6 +126,22 @@ public bool RemoveActivation(ActivationId act, bool force)
             return Instances.Count == 0;
         }
 
+        public IActivationInfo LookupAndRemoveActivation(ActivationAddress address, bool force)
```

Comment:
For the case I implemented, I need to know if an activation was removed, and if so, what was its content. 
I could not see how to do that without adding a new `LookupAndRemove` function on the directory.

The existing `RemoveActivation` does not always do a lookup, and does not return an activation. Instead, it returns a boolean indicating whether there are any activations left after the remove.